### ISSUE TITLE
[Existing device support update]: GE PTAPT-WH02 - expose operating modes through zcl manufacturer-specific cluster

### ISSUE THREAD

**Issue by jsb5151** [Koenkk/zigbee2mqtt#14892]
### Link

https://www.zigbee2mqtt.io/devices/PTAPT-WH02.html

### Database entry

{"id":3,"type":"Router","ieeeAddr":"0x7ce52400000cd0ef","nwkAddr":51676,"manufId":48042,"manufName":"Quirky","powerSource":"Mains (single phase)","modelId":"Smart Switch","epList":[1,2],"endpoints":{"1":{"profId":260,"epId":1,"devId":260,"inClusterList":[0,3,64544],"outClusterList":[6,8,3,25],"clusters":{"genBasic":{"attributes":{"modelId":"Smart Switch","manufacturerName":"Quirky","powerSource":1,"zclVersion":1,"appVersion":6,"stackVersion":2,"hwVersion":1,"dateCode":"20141125"}}},"binds":[],"configuredReportings":[],"meta":{}},"2":{"profId":260,"epId":2,"devId":2,"inClusterList":[0,3,6,5,4],"outClusterList":[],"clusters":{"genBasic":{"attributes":{}},"genOnOff":{"attributes":{"onOff":0}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x00212effff022bb8","endpointID":1}],"configuredReportings":[{"cluster":6,"attrId":0,"minRepIntval":0,"maxRepIntval":3600,"repChange":0}],"meta":{}}},"appVersion":6,"stackVersion":2,"hwVersion":1,"dateCode":"20141125","zclVersion":1,"interviewCompleted":true,"meta":{"configured":1},"lastSeen":1667963975907,"defaultSendRequestWhen":"immediate"}

### Comments

I recently moved from deCONZ to Z2M with good success.  I have 6 of these switches that work just fine, but would like to expose the manufacturer-specific cluster to be able to change the switch to smart bulb mode.

This device is quirky (pun intended) - it has 2 endpoints, 01 is a Dimmer Switch (unused in stock form) and 02 is On/Off Switch.  I kinda got it to work in a very wonky way with deCONZ but it was a PITA since it's not very flexible with level change reports.

Essentially, Endpoint 1's cluster 0xFC20 (dec: 64544) is the one that has the setting; manufactuer-specific code is 0x5488.  I could get it to operate and "lock" the on/off switch by sending a ZCL Attribute change this way in deCONZ CLI: 

```
zclattrmanu 0xabcd 1 0xfc20 0x5488 0200002002

0xABCD being the shortaddr of the switch
1: HA Dimmer endpoint
0xFC20: manufacturer-specific cluster
0x5488: manufacturer code
02: change attribute
0000: attribute 0x0000
20: uint8
02: payload aka config mode
```

Basically, this sets the switch in Smart Bulb mode where it will send Level Control messages from endpoint 1, output cluster 8 (obviously needs reporting or binding - haven't fully figured that one out yet).

So, from what I understand, an update to zigbee-herdsman's cluster.ts (and possibly other files) as well as a customized converter would have to be created, is that correct?  Or can we put everything in a converter?

Thanks.

**Comment by jsb5151** [comment]
Another example of this setting being applied through a SmartThings device handler:

["zcl mfg-code 0x5488", "delay 200",
"zcl global write 0xFC20 0 0x20 {02}", //change the payload to 00,01 for the other modes
"send 0x${device.deviceNetworkId} 1 1","delay 400"]

As well as the original issue I opened on the deCONZ github page w/screenshots of the clusters: https://github.com/dresden-elektronik/deconz-rest-plugin/issues/888
